server/bin/pbench-backup-tarballs.py

```diff
@@ -0,0 +1,381 @@
+                assert os.path.exists(
+                    "{}/{}.md5".format(backup_controller_path, resultname))
+
+                if archive_md5_hex_value == backup_md5_hex_value:
+                    # declare success
+                    nbackup += 1
+                    report.write(
+                        "Already locally backed-up: {}\n".format(resultname))
+                    logger.info(
+                        "Already locally backed-up: {}\n".format(resultname))
+                else:
+                    # quarantine
+                    quarantine(qdir, logger, tb)
+                    nquaran += 1
+                    nerrs += 1
+                    report.write(
+                        "Archive md5 and Backup md5 does not match for: {}\n".format(resultname))
+                    logger.info(
+                        "{} already exists in backup and md5 sums disagree\n".format(resultname))
+                    continue
+        else:
+            md5_done = tar_done = False
+
+            # copy the md5 file from archive to backup
+            try:
+                shutil.copy(archive_md5, backup_controller_path)
+                md5_done = True
+            except Exception:
+                # couldn't copy md5 file
+                nerrs += 1
+                report.write("Failed to copy md5 {} from archive to backup".format(archive_md5))
+                logger.exception(
+                    "shutil.copy: Unable to copy md5 {} from archive to backup: {}\n".format(archive_md5, backup_controller_path))
+
+            # copy the tarball from archive to backup
+            try:
+                shutil.copy(tar, backup_controller_path)
+                tar_done = True
+            except Exception:
+                # couldn't copy tarball
+                nerrs += 1
+                if md5_done:
+                    # remove the copied md5 file from backup
+                    bmd5_file = "{}/{}.md5".format(
+                        backup_controller_path, resultname)
+                    if os.path.exists(bmd5_file):
+                        os.remove(bmd5_file)
+
+            if md5_done and tar_done:
+                nbackup += 1
+                report.write(
+                    "Locally Backed-up Sucessfully: {}\n".format(resultname)
+                )
+                logger.info(
+                    "Locally Backed-up Sucessfully: {}\n".format(resultname))
+
+        # Check if the result already present in s3 or not
+        try:
+            obj = s3_obj.connector.get_object(Bucket='{}'.format(
+                s3_obj.bucket_name), Key='{}'.format(s3_resultname))
+            in_s3 = True
+        except Exception:
+            in_s3 = False
+
+        if in_s3:
+            # compare md5 which we already have so no need to recalculate
+            s3_md5 = obj['ETag'].strip("\"")
+
+            if archive_md5 == s3_md5:
+                # declare success
+                _rename_tb_link(tb, os.path.join(
```

Comment:
I guess the question here is what happens to an already existing link in _linkdest: does this call fail, does it overwrite the link and do we want what it does?